src/plugins/intel_gpu/tests/fake_alignment/fc_fake_alignment_test.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "test_utils.h"
+
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/fully_connected.hpp>
+#include <intel_gpu/primitives/data.hpp>
+
+#include "fully_connected_inst.h"
+
+#include "program_wrapper.h"
+
+#include <cmath>
+#include <algorithm>
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace fake_alignment_tests {
+
+struct fc_fake_align_params {
+    layout input_layout;
+    layout weight_layout;
+    data_types data_type;
+    layout expected_input_layout;
+    layout expected_output_layout;
+};
+
+class fully_connected_fake_align_test : public testing::TestWithParam<fc_fake_align_params> {};
+
+TEST_P(fully_connected_fake_align_test, fake_alignment) {
+    auto p = GetParam();
+
+    auto& engine = get_test_engine();
+
+    auto input_layout_prim = std::make_shared<input_layout>("input", p.input_layout);
+    auto weight_layout_prim = std::make_shared<input_layout>("weight", p.weight_layout);
+    auto fully_connected_prim = std::make_shared<fully_connected>("output", "input", "weight", "", p.data_type);
+
+    cldnn::program prog(engine);
+
+    auto& input_node = prog.get_or_create(input_layout_prim);
+    auto& weight_node = prog.get_or_create(weight_layout_prim);
+    auto& fully_connected_node = prog.get_or_create(fully_connected_prim);
+    program_wrapper::add_connection(prog, input_node, fully_connected_node);
+    program_wrapper::add_connection(prog, weight_node, fully_connected_node);
+
+    auto impl_param = fully_connected_node.get_kernel_impl_params();
+    impl_param->output_layout = fully_connected_inst::calc_output_layouts<ov::PartialShape>(fully_connected_node, *fully_connected_node.get_kernel_impl_params())[0];
+
+    if (impl_param->get_input_layout().is_dynamic() || impl_param->output_layout.is_dynamic()) {
+        EXPECT_THROW(fully_connected_inst::get_fake_aligned_params(*impl_param), std::exception);
+    } else {
+        auto updated_param = fully_connected_inst::get_fake_aligned_params(*impl_param);
```

Comment:
We need to check that the execution is done correctly for fake aligned primitives, so please create few test cases with model execution too.

Reply:
I checked that the new behavior is covered by fc dynamic functional tests.